### PR TITLE
[SPARK-49324] Add `SparkApplication` state transition e2e test

### PR DESCRIPTION
**Pull request by TQJADE**
### What changes were proposed in this pull request?
Add one state transition test for Spark Applications

### Why are the changes needed?
For stablitiy.

### Does this PR introduce _any_ user-facing change?
no
### How was this patch tested?
Test from personal GitHub workflow actions

### Was this patch authored or co-authored using generative AI tooling?
n/a

### CODE REVIEW

File: tests/e2e/python/chainsaw-test.yaml [hunk 1]
```diff
@@ -54,7 +54,7 @@ spec:
         - events:
             namespace: default
         - describe:
-            apiVersion: v1
+            apiVersion: spark.apache.org/v1alpha1
```
Comment (dongjoon-hyun) [inline tests/e2e/python/chainsaw-test.yaml:57]: Why do we need this in this PR, @TQJADE ?

Comment (dongjoon-hyun) [reply tests/e2e/python/chainsaw-test.yaml:57]: I believe this is an orthogonal PR which needs a new JIRA ID.

File: tests/e2e/scripts/verify-state.sh [hunk 2]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
```
Comment (dongjoon-hyun) [inline tests/e2e/scripts/verify-state.sh:1]: Is there a way to avoid this script-based verification ?

**Review by dongjoon-hyun — COMMENTED** [review]
Please rebase this PR to the main branch because #99 is merged .

File: tests/e2e/scripts/verify-state.sh (hunk 2 above)
Comment (TQJADE) [reply tests/e2e/scripts/verify-state.sh]: Updated with assert yaml file

File: tests/e2e/python/chainsaw-test.yaml (hunk 1 above)
Comment (TQJADE) [reply tests/e2e/python/chainsaw-test.yaml]: Removed this from this PR

**Review by dongjoon-hyun — COMMENTED** [review]
It seems that the newly added state transition tests fail. Could you make it pass?

**Comment by TQJADE** [comment]
The local run passed but the CI failed during the last step `kubectl delete sparkapplication $SPARK_APPLICATION_NAME` because of `signal: killed`. Adding timeout to test.

File: tests/e2e/state-transition/chainsaw-test.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    - describe:
+        apiVersion: spark.apache.org/v1alpha1
+        kind: SparkApplication
+        namespace: default
+    finally:
+    - script:
+        env:
+          - name: SPARK_APPLICATION_NAME
+            value: ($SPARK_APPLICATION_NAME)
+        timeout: 30s
```
Comment (TQJADE) [inline tests/e2e/state-transition/chainsaw-test.yaml:51]: Chainsaw default timeout time is 15 seconds: https://kyverno.github.io/chainsaw/latest/configuration/options/timeouts/?h=timeo#supported-timeouts

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @TQJADE .
Merged to main.

**Pull request closed without merging**